### DISCUSSION TITLE
Side-loading calculator

### DISCUSSION THREAD

**Discussion by trombonekenny** [PrairieLearn/PrairieLearn#6993] [Ideas]
I just came from an in-person discussion about PrairieLearn, computer based testing, and calculators. The idea was floated around of including web-based calculators available through PL that could be launched aside a homework or exam. I could see the calculator option being a setting in the assessment where the instructor can decide if it's present, or maybe even the feature set(s) of the calculator.

Is anyone already using a HTML5 or javascript calculator like this? Are there open source copies projects we could install into PL? Or is this something we need/should/could write ourselves?

**Comment by jonatanschroeder** [comment]
Some time ago I was playing around with adding an element with the [desmos calculator](https://app.formative.com/library/5fa2d64c44c26c07ef7a89ff), but I didn't go very far, since the node module I was using had some issues. If anyone is willing to play with it, you'd probably need to either download the files (probably not a good idea from maintainability point of view) or find a suitable node module.

**Reply by rikblok** [reply]
Came across this while Googling.  Here's how I did it in a `question.html` I wrote:

```html
<pl-question-panel>

<div style="float: right;">
    <h5>Optional calculator</h5>
    <iframe src="https://www.desmos.com/fourfunction" width="200" height="350"></iframe>
</div>

<p>
You may not need it in this question, but you are permitted to
use the basic calculator provided to the right.
</p>
</pl-question-panel>
```

**Reply by jonatanschroeder** [reply]
That works fine for homeworks, but may be an issue in exams with controlled Internet resources. Ideally all resources needed during an exam are under a single domain.
